pytest_plugins/settings_skip.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+
+from robottelo.config import setting_is_set
+from robottelo.config import settings
+
+
+def pytest_configure(config):
+    """Register markers related to testimony tokens"""
+    config.addinivalue_line(
+        "markers",
+        'skip_if_not_set: List settings sections that must be set for the test to run. '
+        'If settings are missing, the test is skipped in setup.',
+    )
+
+
+def pytest_runtest_setup(item):
+    """Skip in setup if settings mark isn't met
+
+    settings validate method is used, so required fields are checked
```

Comment:
We'll likely have to make every setting required for this to behave as expected. Otherwise non-required validation won't fail.

Reply:
These are changes that will have to come with the removal of the legacy settings.  I'm retaining the behavior of the current plugin, and this is where lazy validation of dynaconf sections could take place.

Reply:
Sounds good! Dynamic validation should allow us to make everything required, since we're only validating small sections at once.